internal-enrichment/shodan/src/shodanImport.py

```diff
@@ -332,27 +338,50 @@ def _generate_stix_vulnerability(self, data):
 
     def _upsert_stix_observable(self, description, labels, external_reference):
         # Upsert Observable
-        stix_observable = stix2.IPv4Address(
-            id=self.stix_entity["id"],
-            type="ipv4-addr",
-            value=self.stix_entity["value"],
-            custom_properties={
-                "x_opencti_external_references": [external_reference],
-                "x_opencti_description": description,
-                "x_opencti_score": self.score,
-                "x_opencti_labels": labels,
-                "x_opencti_created_by_ref": self.shodan_identity["standard_id"],
-            },
-        )
-        self.stix_objects.append(stix_observable)
-        """
-        {
-                    "source_name": "Shodan",
-                    "url": f"www.shodan.io/host/{data['ip_str']}",
-                    "description": "ceci est une description",
-                    # "external_id": str(data['ip_str']),
-                }
-        """
+        if not self.create_note:
```

Comment:
Here's a small solution for reducing code using dict expansion and ternary:
```python
        # Upsert Observable
        stix_observable = stix2.IPv4Address(
            id=self.stix_entity["id"],
            type="ipv4-addr",
            value=self.stix_entity["value"],
            custom_properties={
                "x_opencti_external_references": [external_reference],
                **({"x_opencti_description": description} if not self.create_note else {}),
                "x_opencti_score": self.score,
                "x_opencti_labels": labels,
                "x_opencti_created_by_ref": self.shodan_identity["standard_id"],
            },
        )
        self.stix_objects.append(stix_observable)
```

Reply:
Good call, I'll make the changes